File: src/main/java/pe/nanamochi/poc_osu_server/packets/client/UserStatsRequestPacket.java
package pe.nanamochi.poc_osu_server.packets.client;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import pe.nanamochi.poc_osu_server.packets.Packet;
import pe.nanamochi.poc_osu_server.packets.Packets;

import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class UserStatsRequestPacket implements Packet {
    private List<Integer> userIds;

    @Override
    public Packets getPacketType() {
        return Packets.OSU_USER_STATS_REQUEST;
    }
}

File: banchus-packet/src/main/kotlin/pe/nanamochi/banchus/packets/client/UserStatsRequestPacket.kt
package pe.nanamochi.banchus.packets.client

import java.io.InputStream
import pe.nanamochi.banchus.core.ClientPacket
import pe.nanamochi.banchus.core.PacketType
import pe.nanamochi.banchus.io.DataReader

class UserStatsRequestPacket : ClientPacket() {
    override val type = PacketType.OSU_USER_STATS_REQUEST

    var userIds: List<Int> = emptyList()

    override fun read(reader: DataReader, ins: InputStream) {
        this.userIds = reader.readIntList16(ins).toList()
    }
}
